Cria o Layout Customizado da collectionView

cs-daniel-lima opened the pull request:
FlowLayout customizado testado e funcionando perfeitamente para alinhar elementos ao lado esquerdo da collectionview

cs-daniel-lima committed:
Improve MagicFlowLayout to align elements to the left side of the UICollectionView
Also adds an extension to UICollectionViewLayoutAttributes to allow the layoutAttribute frame  to have its x position modified in order make sure that the respective cell is aligned left

cs-daniel-lima committed:
Remove APIManagerTests since it is being redone in another branch

cs-kaique-magno reviewed (approved):
Criar tarefa para aplicar testes. Essa tarefa vem tardia devido ao fato da estrutura estar defasada para teste.

cs-daniel-lima commented:
> Criar tarefa para aplicar testes. Essa tarefa vem tardia devido ao fato da estrutura estar defasada para teste.

A tarefa está no board, o MagicFlowLayout está sendo testado junto à CollectionView, a tarefa do board (escrever os testes da collection) poderá ser iniciada assim que conseguirmos testar o APIManager

cs-daniel-lima committed:
Merge branch 'develop'

cs-joao-souza reviewed (changes requested):
Falta teste unitário para a classe.

cs-daniel-lima committed:
Add tests for the MagicFlowLayout class
Add KIF extension
Create MagicFlowLayoutMock and additional Mocks in order to have the view hierarchy necessary to test the MagicFlowLayout

cs-daniel-lima committed:
rename CollectionMockViewController to CollectionViewControllerMock

cs-daniel-lima commented:
@cs-joao-souza testes unitários adicionados

cs-joao-souza commented on magicbyirineuTests/Modules/CardList/MagicFlowLayoutSpec.swift:
```diff
@@ -0,0 +1,93 @@
+        
+        context("when it is initialized"){
+            
+            it("minimumInteritemSpacing should be 0"){
```
```suggestion
            it("has minimumInteritemSpacing equal to 0"){
```

cs-joao-souza replied:
Da uma olhada nos `it`. `has` faz mto mais sentido quando se trata de teste pois should dá uma ideia de que deveria mas n está e o `has` da uma sensação de que realmente tem q ter aquilo

cs-daniel-lima replied:
Resolvido, de fato faz mais sentido usar has. Vlw!

cs-joao-souza commented on magicbyirineuTests/Modules/CardList/MagicFlowLayoutSpec.swift:
```diff
@@ -0,0 +1,93 @@
+        }
+        
+        
+        context("on CollectionViewFlowLayout lifecicle"){
```
`lifecycle` está escrito errado.

cs-joao-souza commented on magicbyirineuTests/Mocks/MagicFlowLayout/CollectionMock.swift:
```diff
@@ -0,0 +1,147 @@
+
+@testable import magicbyirineu
+
+class CollectionViewControllerMock: UIViewController {
```
Gosto de externalizar todas as classes de mocks para q eu saiba onde tem para reutilizar

cs-daniel-lima committed:
Update descriptions on the it statements on MagicFlowLayoutSpec
MagicFlowLayoutMock and CollectionViewControllerMock added to the Mock group

cs-joao-souza commented on magicbyirineuTests/Modules/CardList/MagicFlowLayoutSpec.swift:
```diff
@@ -36,24 +36,24 @@ class MagicFlowLayoutSpec:QuickSpec{
             }
             
-            it("should view match with snapshot when displayed in a UIViewController") {
+            it("should view match with snapshot") {
```
```suggestion
      it("has built layout properly") {
```

cs-daniel-lima replied:
Sugestão incluída no código

cs-joao-souza committed:
Apply suggestions from code review

Co-Authored-By: cs-daniel-lima <47391973+cs-daniel-lima@users.noreply.github.com>

cs-daniel-lima committed:
Idents code properly

cs-daniel-lima committed:
Merge branch 'develop'

Conflicts:
	magicbyirineu.xcodeproj/project.pbxproj

cs-daniel-lima committed:
remove unnecessary cast
The cast is not an optional, so i will aways succeed.
